### PR TITLE
Oauth is now available

### LINKED ISSUE #792: [Enhancement] Oauth client
Could we use https://github.com/knpuniversity/oauth2-client-bundle or similar to login?
I haven't checked yet so I'll ask is this using symfony guard?

In my use case I would need to connect to symfony oauth server to login.

### PR DESCRIPTION
**Pull request by ildyria**
Fixes #792 

Support:
- amazon
- apple
- facebook
- github
- google
- mastodon
- microsoft
- nextcloud  

You will need to modify your `.env` to add your credentials in order to enable Oauth for your instance.

Admin view when Oauth is disabled (the Oauth block will disappear on non admin users):
image

View when all providers are enabled:
image

View when only one provider is enabled:
image

View when only one provider is enabled and registration has been set:
image

Login modal when WebAuthn and GitHub Oauth are enabled:
image

### CODE REVIEW

File: lang/de/oauth.php [hunk 1]
```diff
@@ -0,0 +1,12 @@
...
+
+/** @noinspection SpellCheckingInspection */
+
+return [
+	'NOT_AVAILABLE' => 'Oauth is not available.',
+	'SET_UP_CREDENTIALS' => 'Set up the credentials in your .env',
+	'SET_UP_OAUTH' => 'Set up Oauth authentication',
+	'SET_UP' => 'Set up %s',
+	'TOKEN_REGISTERED' => '%s token registered.',
+	'RESET' => 'reset',
```
Comment (qwerty287) [inline lang/de/oauth.php]:
```suggestion
	'NOT_AVAILABLE' => 'Oauth ist nicht verfügbar.',
	'SET_UP_CREDENTIALS' => 'Richten Sie die Zugangsdaten in Ihrer .env ein',
	'SET_UP_OAUTH' => 'Oauth-Authentifizierung einrichten',
	'SET_UP' => '%s einrichten',
	'TOKEN_REGISTERED' => 'Token %s registriert.',
	'RESET' => 'zurücksetzen',
```

**Review by qwerty287 — APPROVED** [review]
Code lgtm, but I didn't test it.

**Comment by ildyria** [comment]
I still need to add a few tests :)

**Comment by ildyria** [comment]
@qwerty287  Large code change. Could you please double check ? :)

**Review by qwerty287 — APPROVED** [review]
Still looks good I think, but didn't test.

**Pull request merged**